pyrokinetics/gene.py

```diff
@@ -0,0 +1,379 @@
+                ne = species_data.dens
+            else:
+                ion_count += 1
+                name = f'ion{ion_count}'
+
+        pressure = 0.0
+        a_lp = 0.0
+
+        # Normalise to pyrokinetics normalisations and calculate total pressure gradient
+        for name in local_species.names:
+
+            species_data = local_species[name]
+
+            species_data.temp = species_data.temp / te
+            species_data.dens = species_data.dens / ne
+            
+            pressure += species_data.temp * species_data.dens
+            a_lp += species_data.temp * species_data.dens * (species_data.a_lt + species_data.a_ln)
+
+            species_data.name = name
+
+            # Add individual species data to dictionary of species
+            local_species[name] = species_data
+            local_species.names.append(name)
+
+        local_species.pressure = pressure
+        local_species.a_lp = a_lp
+
+        # Add local_species
+        pyro.local_species = local_species
+
+    def pyro_to_gene_miller(self):
+        """
+        Generates dictionary of equivalent pyro and gene parameter names
+        for miller parameters
+        """
+
+        pyro_gene_param = {
+            'rho' : ['geometry', 'minor_r'],
+            'Rmaj' : ['geometry', 'major_r'],
+            'Rgeo' : ['geometry', 'major_r'],
+            'q' : ['geometry', 'q0'],
+            'kappa' : ['geometry', 'kappa'],
+            's_kappa' : ['geometry', 's_kappa'],     #checked
+            'delta' : ['geometry', 'delta'],
+            's_delta' : ['geometry', 's_delta'],    #checked
+            'shat' : ['geometry', 'shat'],  
+            'shift' : ['geometry', 'drr'],
+            }
+
+        return pyro_gene_param
+
+    def pyro_to_gene_species(self):
+        """
+        Generates dictionary of equivalent pyro and gene parameter names
+        for species parameters
+        """
+
+        pyro_gene_species = {
+            'mass' : 'mass',
+            'z' :  'charge',
+            'dens': 'dens',
+            'temp': 'temp',
+            'a_lt' : 'omt',
+            'a_ln' : 'omn',
+            }
+
+        return pyro_gene_species
+
+    def add_flags(self, pyro, flags):
+        """
+        Add extra flags to GENE input file
+
+        """
+
+        for key, parameter in flags.items():
+            for param, val in parameter.items():
+
+                pyro.gene_input[key][param] = val
+
+
+    def load_numerics(self, pyro, gene):
+        """
+        Load GENE numerics into Pyro
+
+        """
+        # Need shear for map theta0 to kx
+        # shat = pyro.miller['shat']
+        # Fourier space grid
+        # Linear simulation
+        
+        numerics = Numerics()
+
+        numerics['nky'] = gene['box']['nky0']
+        numerics['nkx'] = gene['box']['nx0']
+        numerics['nky'] = 1
+        numerics['ky'] = gene['box']['kymin'] 
+        numerics['kx'] = 1.0
+
+
+        # Velocity grid
```

Comment:
Do we need to set nz0, nw0 and nv0 somehow? 

I would imagine 
```python
nz0 = numerics.ntheta
nv0 = 2 * numerics.nenergy
nw0 = numerics.npitch
```
 would be reasonable? Thoughts?

Reply:
Probably a good idea to have a sensible default. These seem reasonable to me. 

Was hard to think of something here on the basis that I can't personally imagine not just fiddling with these myself, but no harm in having something to start with.

Reply:
Agreed, we can always change it as we moved forward

Reply:
Now done.